src/ShareCreated.php

```diff
@@ -37,14 +38,14 @@ public function getPermissionId(): string
      */
     public function setRole(SharingRole $role): bool
     {
-        $this->apiPermission->setRoles([$role->getId()]);
-
+        $apiPermission = new ApiPermission();
+        $apiPermission->setRoles([$role->getId()]);
```

Comment:
why do we need this change?

Reply:
`$this->apiPermission` contains unnecessary values and `updatePermission` API endpoint body doesn't support those values so initiating new apiPermission which contain role key-value pair